[TSP Migration]--elastic

welovej opened the pull request:
This PR migrates your latest version (identified by the tag in your readme.md) of swagger to TypeSpec. We already tried our best to make sure the TypeSpec represents same as previous swagger. Since we lack the business knowledge, please validate this PR again to make sure it's functional equivalent as before. The local validation step is at [Getting started | TypeSpec Azure](https://azure.github.io/typespec-azure/docs/migrate-swagger/01-get-started/#review-and-adjust-the-typespec)

Besides, TypeSpec encourages to follow [ARM guidelines](https://aka.ms/rpguidelines). Therefore, some representations in your previous swagger will be fixed to follow these guidelines. When you see differences in your local validation, please keep this note in mind.

Please reach out to [TypeSpec Discussions Channel](https://teams.microsoft.com/l/channel/19%3a906c1efbbec54dc8949ac736633e6bdf%40thread.skype/TypeSpec%2520Discussion%2520%25F0%259F%2590%25AE?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) if there is any help needed.

welovej committed:
newly converted

welovej committed:
compiled

welovej committed:
Merge remote-tracking branch 'upstream/main' into TspMig-elastic

welovej committed:
update

welovej committed:
formatted

welovej committed:
prettier

welovej committed:
suppress some warnings

welovej committed:
Update models.tsp

welovej committed:
update

welovej committed:
Merge remote-tracking branch 'upstream/main' into TspMig-elastic

JiaqiZhang-Dev commented:
/azp run SDK Validation - Go

welovej committed:
examples update

welovej committed:
optional body fix and other diff fix

welovej committed:
justification

welovej committed:
remove previous api version example files

welovej committed:
recovery of previous api version files

welovej committed:
Fix addtional model  Azure.ResourceManager.ArmResponse<ElasticMonitorResource>

XiaofeiCao committed:
Update java config

XiaofeiCao commented on specification/elastic/Elastic.Management/main.tsp:
```diff
@@ -0,0 +1,71 @@
+  v2025_06_01: "2025-06-01",
+}
+
+/**
+ * This is the interface that implements the standard Azure Resource Manager operation that returns
+ * all supported RP operations. You should have exactly one declaration for each
+ * Azure Resource Manager service. It implements
+ *   GET "/providers/Microsoft.ContosoProviderHub/operations"
+ */
+interface Operations {
```
Why not use the standard `Azure.ResourceManager.Operations`?

welovej committed:
update

v-jiaodi committed:
Merge branch 'main' into TspMig-elastic

welovej committed:
update

welovej committed:
Merge branch 'TspMig-elastic' of https://github.com/welovej/azure-rest-api-specs into TspMig-elastic

welovej committed:
prettier

melina5656 committed:
Remove Bodyboot Decorator

v-jiaodi committed:
resolve breaking

v-jiaodi committed:
Merge branch 'main' into TspMig-elastic

v-jiaodi committed:
Merge branch 'main' into TspMig-elastic

v-jiaodi commented on specification/elastic/Elastic.Management/ElasticMonitorResource.tsp:
```diff
@@ -0,0 +1,498 @@
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @action("listAllTrafficFilters")
+  @operationId("AllTrafficFilters_list")
```
It should be a separate operation group and should not be included in `ElasticMonitorResources`.

Commit:
update

Commit:
fix go breaking

welovej commented:
> Lack of `@clientLocation` customization.

Added

ArthurMa1978 committed:
Merge branch 'main' into TspMig-elastic

welovej committed:
fix as description

System.ArgumentException: 'An item with the same key has already been added. Key: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}'

welovej committed:
update

welovej committed:
update

welovej committed:
fix OkResponse issue for SDK generation

welovej committed:
update

weidongxu-microsoft committed:
Java SDK migration: Fix createAndAssociateIpFilters operation name for backward compatibility

HarveyLink commented on specification/elastic/Elastic.Management/models.tsp:
```diff
@@ -0,0 +1,1347 @@
+  /**
+   * Provisioning state of the monitoring tag rules.
+   */
+  @visibility(Lifecycle.Read)
```
Please check if it's readonly.

welovej committed:
Update tspconfig.yaml

ArthurMa1978 committed:
Merge branch 'main' into TspMig-elastic

ArthurMa1978 committed:
update client settings

welovej committed:
Update main.tsp

welovej committed:
Merge branch 'main' into TspMig-elastic

welovej committed:
Update models.tsp

welovej committed:
update

welovej committed:
Update tspconfig.yaml

pshao25 committed:
Merge branch 'main' into TspMig-elastic
